src/Core/Services/RestService.cs

```diff
@@ -431,8 +437,15 @@ public string GetBaseRouteFromConfig()
         /// (and optionally primary key).</param>
         /// <returns>entity name associated with entity path and primary key route.</returns>
         /// <exception cref="DataApiBuilderException"></exception>
-        public (string, string) GetEntityNameAndPrimaryKeyRouteFromRoute(string routeAfterPathBase)
+        public (string, string) GetEntityNameAndPrimaryKeyRouteFromRoute(string routeAfterPathBase, string dataSourceName = "")
```

Comment:
update method comment to include dataSourceName param

Reply:
Rest should use default datasourceName as this pr is not handling multi-source support for rest. making that change.